src/miniso.rs

```diff
@@ -66,7 +66,9 @@ impl Table {
             });
         }
 
-        entries.sort_by_key(|e| e.minimal.as_sector());
+        // empty files can share an address with a nonempty file, so make
+        // sure we see the empty ones first
+        entries.sort_by_key(|e| (e.minimal.as_sector(), e.length));
```

Comment:
Hmm, let's just delete it from the table instead? It doesn't hurt the algorithm but it has no value for our purposes and takes up space in the table. We should still make sure it's accounted for in the `hardlinks` count though (which should be renamed e.g. `extraneous` or something) for reporting.

Reply:
> but it has no value for our purposes

Well, I guess it tests that we handle the 0 case well... So meh, not sure. Would probably still nuke it though. Sounds more like a job for unit tests.

Reply:
Sounds good; I've done that now.  While testing that change, I caught genisoimage putting empty files at some random offset into another file (!) so the original fix wouldn't have been sufficient anyway.